std/file.d

```diff
@@ -69,6 +69,12 @@ package @property string deleteme() @safe
     return _deleteme;
 }
 
+version (unittest) private struct TestAliasedString
+{
+    string s;
+    alias s this;
+}
```

Comment:
You should add `@disable this(this);` to the test struct to ensure that the struct is not actually copied.

In addition, DirEntry aliases to an r-value, which caused additional problems, so it's probably better to alias this to a property function return `s`;